lightningd/plugin.c

```diff
@@ -415,10 +416,33 @@ static void plugin_rpcmethod_destroy(struct json_command *cmd,
 static void plugin_rpcmethod_cb(const struct plugin_request *req,
 				struct plugin_rpc_request *rpc_req)
 {
-	// Parse
-	// Extract results or error
-	// Construct reply
-	// Return result with appropriate return code.
+	struct json_stream *response;
+	const jsmntok_t *res;
+	assert(req->resulttok || req->errortok);
+
+	if (req->errortok) {
+		res = req->errortok;
+		command_fail(rpc_req->cmd, PLUGIN_ERROR, "%.*s",
+			     res->end - res->start, req->response + res->start);
+		tal_free(rpc_req);
+		return;
+	}
+
+	res = req->resulttok;
+	response = json_stream_success(rpc_req->cmd);
+
+	/* If this is a raw string result we'd be cropping the quotes from the
+	 * result, so add them back */
+	if (req->resulttok->type == JSMN_STRING)
+		json_add_member(response, NULL, "\"%.*s\"",
+				res->end - res->start,
+				req->response + res->start);
+	else
+		json_add_member(response, NULL, "%.*s", res->end - res->start,
+				req->response + res->start);
```

Comment:
BTW, there are `json_tok_contents` and `json_tok_len` for exactly this "strings are special" case:

```
     json_add_member(response, NULL, "%.*s", json_tok_len(t), json_tok_contents(t));
```

Reply:
Thanks, will do a pass and replace where necessary.

Reply:
Turns out this is the only case this is necessary, so I added a fixup :+1: